### PR TITLE
Codeforafrica Stories

### LINKED ISSUE #592: @codeforafrica Stories
Implement Stories and individual stories pages

The implementation should also keep in mind https://github.com/CodeForAfrica/ui/pull/443 since this approach promises to minimise changes should content editors move blocks around.

### PR DESCRIPTION
**Pull request by kelvinkipruto**
## Description
This PR implements creating and rendering the list of stories.

Fixes #592 

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Screenshots
image

image

image

image

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/codeforafrica/src/components/ArticleCardList/ArticleCardList.js [hunk 1]
```diff
@@ -37,15 +37,15 @@ const ArticleCardList = React.forwardRef(function ArticleCardList(props, ref) {
                   display: { xs: "none", sm: "block" },
                   height: "217.64px",
                 }}
-                src={article.featureImage}
+                src={article.image.src}
```
Comment (koechkevin) [inline apps/codeforafrica/src/components/ArticleCardList/ArticleCardList.js:40]: @kelvinkipruto can we guarantee that image can never be null or undefined?

**Review by koechkevin — COMMENTED** [review]
@kelvinkipruto Are we not supposed to implement filters on this PR.  Clicking tags on preview app doesn't seem to work.

File: apps/codeforafrica/src/components/ArticleCardList/ArticleCardList.js (hunk 1 above)
Comment (kelvinkipruto) [reply apps/codeforafrica/src/components/ArticleCardList/ArticleCardList.js:40]: @koechkevin Yes. The cover image is a required field

File: apps/codeforafrica/src/lib/data/blockify/stories.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  } = await api.getCollection("article", options);
+
+  const uniqueTags = new Set(
+    storyList
+      .reduce((acc, story) => {
+        const { tags = [] } = story;
+        return [...acc, ...tags.map((tag) => tag.name)];
+      }, [])
+      .sort(),
+  );
```
Comment (kilemensi) [inline apps/codeforafrica/src/lib/data/blockify/stories.js:60]: Similar to the members PR, we need to think about how are we going to get tags for all stories & not just the one visible on the current page

File: apps/codeforafrica/src/payload/fields/articles.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { deepmerge } from "@mui/utils";
+
+const articles = (overrides) =>
```
Comment (kilemensi) [inline apps/codeforafrica/src/payload/fields/articles.js:3]: Is this being reused anywhere else?

**Review by kilemensi — COMMENTED** [review]
👍🏽 

----

There are so many changes on `Articles` component. What are they all about?

**Comment by kelvinkipruto** [comment]
> There are so many changes on `Articles` component. What are they all about?

@kilemensi My initial plan was to split this task into 3 PR, this one for rendering all the stories, pagination in #608 and filtering in #609. 
I think I should merge all of them into this PR as the changes are not as much as I had thought.

**Review by koechkevin — APPROVED** [review]

**Review by kilemensi — COMMENTED** [review]
👍🏽 ... Just had a quick look on the CMS side:

1. Lets arrange the page to resemble other CMS. Main part of the page should contain cover image, title, and content only. Side panel should contain slug, publish date, tags, excerpt (textarea and not richText), and author.
2. Aren't we keeping the old architecture of having just **Posts** collection with **Stories** and **Opportunities** using different tags?
3. Posts should also have SEO like pages

**Review by koechkevin — APPROVED** [review]

**Pull request merged**